torchci/pages/api/drci/drci.ts

```diff
@@ -40,26 +51,104 @@ export async function updateDrciComments(octokit: Octokit, prNumber?: string) {
         NUM_MINUTES + ""
     );
 
-    const workflowsByPR = reorganizeWorkflows(recentWorkflows);
+    const workflowsByPR = await reorganizeWorkflows(octokit, recentWorkflows);
     const sevs = getActiveSEVs(await fetchIssuesByLabel("ci: sev"));
+    const masterFlakyJobs = await getFlakyJobs();
+    const baseCommitJobs = await getBaseCommitJobs(workflowsByPR);
 
     for (const [pr_number, pr_info] of workflowsByPR) {
-        const { pending, failedJobs } = getWorkflowJobsStatuses(pr_info);
+        const { pending, failedJobs, flakyJobs, brokenTrunkJobs } =
+          getWorkflowJobsStatuses(
+            pr_info,
+            masterFlakyJobs,
+            baseCommitJobs.get(pr_info.merge_base) || new Map()
+          );
 
-        const failureInfo = constructResultsComment(pending, failedJobs, pr_info.head_sha);
-        const comment = formDrciComment(pr_number, failureInfo, formDrciSevBody(sevs));
+        const failureInfo = constructResultsComment(
+          pending,
+          failedJobs,
+          flakyJobs,
+          brokenTrunkJobs,
+          pr_info.head_sha,
+          pr_info.merge_base
+        );
+        const comment = formDrciComment(
+          pr_number,
+          failureInfo,
+          formDrciSevBody(sevs)
+        );
 
         await updateCommentWithWorkflow(octokit, pr_info, comment);
     }
+    console.log("done")
+}
+
+
+async function getBaseCommitJobs(
+  workflowsByPR: Map<number, PRandJobs>
+): Promise<Map<string, Map<string, RecentWorkflowsData>>> {
+  // get merge base shas
+  let baseShas = [];
+  for (const [_, pr_info] of workflowsByPR) {
+    baseShas.push(pr_info.merge_base);
```

Comment:
might this introduce duplicate entries?

Reply:
yes, but the rockset query should handle it ok

Reply:
Good to know, though it might still cause extra perf overhead. Consider replacing this array with a [set](https://www.javatpoint.com/typescript-set#:~:text=TypeScript%20set%20is%20a%20new,not%20the%20key%2Dvalue%20pairs.) or something similar